meson.build

```diff
@@ -61,6 +61,34 @@ executable(
 	],
 	include_directories: [inc],
 	install: true,
+	install_dir: get_option('libexecdir'),
+)
+
+conf_data = configuration_data()
+conf_data.set('libexecdir',
+	join_paths(get_option('prefix'), get_option('libexecdir')))
+conf_data.set('systemd_service', '')
+
+systemd = dependency('systemd', required: get_option('systemd'))
+
+if systemd.found()
+	systemd_service_file = 'xdg-desktop-portal-wlr.service'
+	user_unit_dir = systemd.get_pkgconfig_variable('systemduserunitdir')
```

Comment:
Can we add `define_variable: ['prefix', get_option('prefix')]` to respect the prefix set by the user here?

Reply:
We can, but that's probably not going to work.

There is a fixed list of paths that systemd scans for service files (`systemduserunitpath` pkgconfig variable + matching paths under $HOME) and if the user does something like `--prefix=/app/whatever-path/` the service won't be picked up.

Reply:
We can't install anything outside of the prefix. For instance installing to `~/.local` shouldn't require any root permissions. It's up to the user to setup `$SYSTEMD_UNIT_PATH` correctly to make systemd pick up the unit files.

Reply:
Done. I was concerned about nixos, but it seems that they ignore upstream service files and generate their own.
This could be done smarter (i.e. for `~/.local` correct path for user units is `~/.local/share/systemd/user` instead of `~/.local/lib/systemd/user`), but all the branching logic would have to be defined in `meson.build` which is an extra maintenance burden.